udev: substitute /usr/bin/cat in rules

cmm opened the pull request:
alsa-utils 1.2.13 introduced a change to a udev rules file that references /usr/bin/cat, so let's substitute that too.

avoids NixOS build failure with "/usr/bin/cat is called in udev rules but is not executable or does not exist".

cmm committed:
udev: substitute /usr/bin/cat in rules

Used by alsa-utils 1.2.13.

lucasew commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 370740`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>tests.devShellTools.nixos</li>
    <li>tests.testers.lycheeLinkCheck.network</li>
    <li>tests.testers.nixosTest-example</li>
    <li>tests.testers.runNixOSTest-example</li>
    <li>tests.trivial-builders.references</li>
  </ul>
</details>

Shawn8901 commented on nixos/modules/services/hardware/udev.nix:
```diff
@@ -66,6 +66,7 @@ let
           --replace \"/sbin/blkid \"${pkgs.util-linux}/sbin/blkid \
           --replace \"/bin/mount \"${pkgs.util-linux}/bin/mount \
           --replace /usr/bin/readlink ${pkgs.coreutils}/bin/readlink \
+          --replace /usr/bin/cat ${pkgs.coreutils}/bin/cat \
```
AFAIK `--replace` is deprecated ( https://github.com/NixOS/nixpkgs/issues/356002 ), could we switch to `--replace-quiet`? That should be the same behavior, but non deprecated. If I understand the code snippet correct we are traversing all kind of possible udev rules, so no one knows if all replacements are existing.

cmm replied:
that is a valid point, but should it be addressed in this PR?

Shawn8901 replied:
For me it's valid to do some chore work aside touching code, tho I would place it in a separate commit.
But I am also just a random person :)

cmm replied:
fair enough, actually :)

cmm committed:
udev: "--replace" -> "--replace-quiet" in substitutions

"--replace" is deprecated, and neither vocal new alternative is
appropriate in this case.
